src/index.js

```diff
@@ -1088,6 +1147,115 @@ class MiniCssExtractPlugin {
+                          `if (${RuntimeGlobals.scriptNonce}) {`,
+                          Template.indent(
+                            `link.setAttribute("nonce", ${RuntimeGlobals.scriptNonce});`
+                          ),
+                          "}",
+                          'link.rel = "prefetch";',
+                          'link.as = "style";',
+                          `link.href = ${RuntimeGlobals.publicPath} + ${RuntimeGlobals.require}.miniCssF(chunkId);`,
+                        ]),
+                        chunk
+                      ),
+                      "document.head.appendChild(link);",
+                    ]),
+                    `}`,
+                  ]),
+                  "}",
+                ])};`
+              : "// no prefetching",
+            "",
+            withPreload && hasCssMatcher !== false
+              ? `${
+                  RuntimeGlobals.preloadChunkHandlers
+                }.miniCss = ${runtimeTemplate.basicFunction("chunkId", [
+                  `if((!${
+                    RuntimeGlobals.hasOwnProperty
+                  }(installedCssChunks, chunkId) || installedCssChunks[chunkId] === undefined) && ${
+                    hasCssMatcher === true ? "true" : hasCssMatcher("chunkId")
+                  }) {`,
+                  Template.indent([
+                    "installedCssChunks[chunkId] = null;",
+                    linkPreload.call(
+                      Template.asString([
+                        "var link = document.createElement('link');",
+                        "link.charset = 'utf-8';",
+                        `if (${RuntimeGlobals.scriptNonce}) {`,
+                        Template.indent(
+                          `link.setAttribute("nonce", ${RuntimeGlobals.scriptNonce});`
+                        ),
+                        "}",
+                        'link.rel = "preload";',
+                        'link.as = "style";',
+                        `link.href = ${RuntimeGlobals.publicPath} + ${RuntimeGlobals.require}.miniCssF(chunkId);`,
+                        crossOriginLoading
+                          ? crossOriginLoading === "use-credentials"
+                            ? 'link.crossOrigin = "use-credentials";'
+                            : Template.asString([
+                                "if (link.href.indexOf(window.location.origin + '/') !== 0) {",
+                                Template.indent(
+                                  `link.crossOrigin = ${JSON.stringify(
+                                    crossOriginLoading
+                                  )};`
+                                ),
+                                "}",
+                              ])
+                          : "",
+                      ]),
+                      chunk
+                    ),
+                    "document.head.appendChild(link);",
+                  ]),
+                  "}",
+                ])};`
+              : "// no preloaded",
```

Comment:
Did you copy this code from webpack?

Reply:
Yes, with adaptations to CSS.
Except for the `getLinkElements` function that I added (https://github.com/webpack-contrib/mini-css-extract-plugin/pull/1043/commits/b785fd8eff6e09073f5d79f7dae498a3a2f62c32).

Reply:
Yes, with adjustments to CSS.
Except for the `getLinkElements` function that I added (https://github.com/webpack-contrib/mini-css-extract-plugin/pull/1043/commits/b785fd8eff6e09073f5d79f7dae498a3a2f62c32).

Reply:
I want to finish - https://github.com/webpack/webpack/pull/18190/, i.e. for built-in CSS support, then I will check code here, regaridng - `getLinkElements`, it is an intresting question, because I think browsers make different things - https://github.com/webpack/webpack/pull/17497, we need to report about to chrome bug tracker, if it is a bug, they need to fix it, if not - I will improve our code late